### PR TITLE
Martyr's mail

### PR DESCRIPTION
**Pull request by Declow0**
Issue #121.

Implementation of Martyr's mail. Healing doesn't work on caster.

### CODE REVIEW

File: game/scripts/vscripts/items/martyrs_mail.lua [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+	return self.bonus_intellect
+end
+
+--------------------------------------------------------------------------------
+
+modifier_item_martyrs_mail_martyr_active = class({})
+
+function modifier_item_martyrs_mail_martyr_active:IsAura()
+	if self:GetCaster() == self:GetParent() then
+		return true
```
Comment (DoctorGester) [inline game/scripts/vscripts/items/martyrs_mail.lua]: Why? Just return true should work, no?

File: game/scripts/vscripts/items/martyrs_mail.lua [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+function modifier_item_martyrs_mail_martyr_active:GetEffectAttachType()
+	return PATTACH_ABSORIGIN_FOLLOW
+end
+
+--------------------------------------------------------------------------------
+
+modifier_item_martyrs_mail_martyr_aura = class({})
+
+function modifier_item_martyrs_mail_martyr_aura:IsHidden()
+	if self:GetCaster() == self:GetParent() then
```
Comment (DoctorGester) [inline game/scripts/vscripts/items/martyrs_mail.lua]: GetAuraEntityReject?

File: game/scripts/vscripts/items/martyrs_mail.lua (hunk 1 above)
Comment (Declow0) [inline game/scripts/vscripts/items/martyrs_mail.lua]: Yes, this part doesn't need.

Comment (Declow0) [inline game/scripts/vscripts/items/martyrs_mail.lua]: Works return and it provide visual aura buff. Done.

File: game/scripts/vscripts/items/martyrs_mail.lua (hunk 2 above)
Comment (Declow0) [inline game/scripts/vscripts/items/martyrs_mail.lua]: Replace by GetAuraEntityReject. Done.

File: game/scripts/vscripts/items/martyrs_mail.lua [hunk 3]
```diff
@@ -75,17 +75,21 @@ end
...
-	return false
+	return true
 end
 
 function modifier_item_martyrs_mail_martyr_active:GetModifierAura()
 	return "modifier_item_martyrs_mail_martyr_aura"
 end
 
+function modifier_item_martyrs_mail_martyr_active:GetAuraEntityReject( hEntity )
+	if self:GetCaster() == hEntity then
```
Comment (DoctorGester) [inline game/scripts/vscripts/items/martyrs_mail.lua]: return self:GetCaster() == hEntity

File: game/scripts/vscripts/items/martyrs_mail.lua [hunk 4]
```diff
@@ -151,14 +155,6 @@ end
...
 
-function modifier_item_martyrs_mail_martyr_aura:IsHidden()
-	if self:GetCaster() == self:GetParent() then
-		return true
-	end
-	
-	return false
-end
-
 function modifier_item_martyrs_mail_martyr_aura:GetEffectName()
```
Comment (DoctorGester) [inline game/scripts/vscripts/items/martyrs_mail.lua]: Should just return an effect name, condition isn't required anymore

**Pull request merged**